lib/jxl/enc_frame.cc

```diff
@@ -1167,9 +1163,27 @@ Status EncodeGlobalACInfo(PassesEncoderState* enc_state, BitWriter* writer,
     if (enc_state->cparams.decoding_speed_tier >= 1) {
       hist_params.max_histograms = 6;
     }
+    size_t num_histogram_groups = shared.num_histograms;
+    if (enc_state->streaming_mode) {
+      size_t prev_num_histograms =
+          enc_state->passes[i].codes.encoding_info.size();
+      if (enc_state->update_global_state) {
+        prev_num_histograms += kNumFixedHistograms;
+        hist_params.add_fix_histograms = true;
+      }
+      size_t remaining_histograms = kClustersLimit - prev_num_histograms;
+      size_t max_histograms = remaining_histograms < 20
```

Comment:
I assume this is a heuristic and the constants could be tuned in the future :)

Reply:
Yes, added a comment and a TODO